recipes-ni/ni-shutdown-guard/ni-shutdown-guard_1.0.bb

```diff
@@ -0,0 +1,30 @@
+SUMMARY = "Shutdown/reboot guard utility for NILRT"
+DESCRIPTION = "Utility to prevent shutdown/reboot to protect critical operations"
+LICENSE = "MIT"
+LIC_FILES_CHKSUM = "file://${COMMON_LICENSE_DIR}/MIT;md5=0835ade698e0bcf8506ecda2f7b4f302"
+SECTION = "base"
+
+SRC_URI = "\
+	file://rguard \
+	file://holdoff-shutdown \
+"
+
+FILES_${PN} += "\
+	${sbindir}/rguard \
+	${sysconfdir}/init.d/holdoff-shutdown \
+"
+
+DEPENDS += "shadow-native pseudo-native update-rc.d-native"
+
+RDEPENDS_${PN} += "bash"
+
+do_install () {
+	install -d ${D}${sbindir}
+	install -d ${D}${sysconfdir}/init.d
+	install -d ${D}${sysconfdir}/holdoff-shutdown.d
+
+	install -m 0755   ${WORKDIR}/rguard              ${D}${sbindir}
+	install -m 0755   ${WORKDIR}/holdoff-shutdown    ${D}${sysconfdir}/init.d
+
+	update-rc.d -r ${D} holdoff-shutdown      stop 00 0 6 .
```

Comment:
Could you use the [`update-rc.d.bblclass`](http://www.embeddedlinux.org.cn/OEManual/update-rc-d_class.html) instead?